drivers/SmartThings/jbl/src/discovery_mdns.lua

```diff
@@ -0,0 +1,150 @@
+local log = require "log"
+local mdns = require "st.mdns"
+local net_utils = require "st.net_utils"
+
+local discovery_mdns = {}
+
+local function byte_array_to_plain_text(byte_array)
```

Comment:
Not as concerned about the ASCII stuff, but I would advocate for simplifying this to use as much Lua built-in functionality as possible. You can replace this with:

```lua
string.char(table.unpack(byte_array))
```

[Lua's `string.char` function is variadic](https://www.lua.org/manual/5.0/manual.html#5.3).